src/components/color_picker/color_picker_entry.ts

```diff
@@ -0,0 +1,45 @@
+import { Component } from "@odoo/owl";
+import { SpreadsheetChildEnv } from "../../types";
+import { css } from "../helpers";
+import { ColorPicker } from "./color_picker";
+
+interface Props {
+  currentColor: string | undefined;
+  onClick: () => void;
+  showColorPicker: boolean;
+  onColorPicked: (color: string) => void;
+  icon: string;
+  dropdownDirection?: "left" | "right" | "center";
+  tooltipText?: string;
+}
+
+css/* scss */ `
+  .o-color-picker-entry span {
+    cursor: pointer;
+  }
+
+  .o-cf-cell-is-rule .o-color-picker-entry span {
+    cursor: default;
+  }
+`;
+
+export class ColorPickerEntry extends Component<Props, SpreadsheetChildEnv> {
```

Comment:
Not a fan of the name `ColorPickerEntry`, that being said I don't have any bright idea. `ColorPickerContainer` ? `ColorPickerButton` ?

Reply:
Let's wait for others' ideas